modules/geolocation_log_to_kml.py

```diff
@@ -0,0 +1,116 @@
+"""
+Convert log file to KML file.
+"""
+
+import argparse
+import pathlib
+import re
+
+from .common.modules import location_global
+from .common.modules import location_local
+from .common.modules import position_global
+from .common.modules.kml import kml_conversion
+from .common.modules.mavlink import local_global_conversion
+
+
+DEFAULT_RESULTS_PATH = pathlib.Path("results")
+
+
+# pylint: disable=redefined-outer-name
+def convert_geolocation_log_to_kml(
+    log_file: str,
+    home_position: position_global.PositionGlobal,
+    document_name_prefix: str,
+    save_directory: str,
+) -> "tuple[bool, pathlib.Path | None]":
+    """
+    Given a geolocation log file with a specific format, return a corresponding KML file.
+
+    Args:
+        log_file (str): Path to the geolocation log file
+        document_name_prefix (str): Prefix name for saved KML file.
+        save_directory (str): Directory to save the KML file to.
+
+    Returns:
+        tuple[bool, pathlib.Path | None]: Returns (False, None) if function
+            failed to execute, otherwise (True, path) where path a pathlib.Path
+            object pointing to the KML file.
+    """
+    locations = []
+
+    with open(log_file, "r", encoding="utf-8") as f:
+        for line in f:
+            # find all the points within the line (should be just 1)
+            points = re.findall(
+                r"centre: \[.*\]", line
+            )  # fornmat: [ 'centre:', '[', '123.123', '123.123]' ]
+
+            if len(points) > 0:
+                point = points[0].split()
+                north = float(point[2])
+                east = float(point[3][:-1])
```

Comment:
Why is this different from north?

Reply:
Hi @maxlou05, I also need the geolocation log file format to address this issue here. But for now, I'm assuming that `point[3]` also includes a trailing character such as ",", ")" or something else?